test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1606,3 +1606,312 @@ class instead of the class it was in when it was queued.
+
+        self.server.holdjob(jid1, USER_HOLD)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+
+        self.scheduler.log_match("Number of job equivalence classes: 1",
+                                 max_attempts=10, starttime=self.t)
+
+    def test_suspended_job_node(self):
+        "Test that a suspended job goes back to the same node on re queue"
+
+        a = {'resources_available.ncpus': 1}
+        self.server.create_vnodes('vnode', a, 2, self.mom, usenatvnode=True)
+
+        a = {'Resource_List.ncpus': 1}
+        J = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(J)
+
+        # create express queue
+        a = {'queue_type': 'e', 'started': 't',
+             'enabled': 't', 'priority': 150}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='expressq')
+
+        a = {'Resource_List.ncpus': 1, 'queue': 'expressq'}
+        Je = Job(TEST_USER, attrs=a)
+        jid2 = self.server.submit(Je)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        time.sleep(2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+
+        self.scheduler.log_match("Number of job equivalence classes: 1",
+                                 max_attempts=10, starttime=self.t)
+
+        a = {'Resource_List.ncpus': 1, 'queue': 'expressq'}
+        Je = Job(TEST_USER, attrs=a)
+        jid3 = self.server.submit(Je)
+
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+
+        # 1 equi class for j1d, 1 for jid2 and jid3
+        self.scheduler.log_match("Number of job equivalence classes: 2",
+                                 max_attempts=10, starttime=self.t)
+
+        # Delete jid2 and check that jid1 is still in suspended state
+        self.server.deljob(jid2, wait=True)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+
+        # delete jid3 and check that jid1 goes back to R state
+        self.server.deljob(jid3, wait=True)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.scheduler.log_match("Number of job equivalence classes: 1",
+                                 max_attempts=10, starttime=self.t)
+
+    def test_multiple_node_multiple_job_order(self):
+        """Test jobs that come later in query order on multi-node
+        setup are added to its own equivalence class.This is
+        multinode test for test_premption2"""
+
+        # create 2 nodes
+        a = {'resources_available.ncpus': 3}
+        self.server.create_vnodes('vn', a, 2, self.mom, usenatvnode=False)
+
+        # create express queue
+        q = {'queue_type': 'e', 'started': 't',
+             'enabled': 't', 'priority': 150}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, q, id='expressq')
+
+        # start 2 jobs of same equivalence class on node 1
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=1:vnode=vn[0]'})
+        jid1 = self.server.submit(j1)
+
+        time.sleep(1)
```

Comment:
The sleep probably deserves a comment explaining it.  The reason it is here is because the scheduler will preempt jobs starting with the ones which were most recently started.  The granularity is in seconds.  This means if two jobs are run in the same second, the scheduler might choose either one.  By sleeping one second between submitting the two jobs, it forces the scheduler to preempt the second one.